amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestConsumer.java

```diff
@@ -220,4 +258,53 @@ private void deleteResources(StreamExistenceManager streamExistenceManager, Leas
         log.info("-------------Finished deleting resources.----------------");
     }
 
+    @Data
+    @ToString
+    @AllArgsConstructor
+    public static class StreamScaler implements Runnable {
+        private final KinesisAsyncClient client;
+        private final String streamName;
+        private final Queue<ReshardOptions> scalingFactor;
+        private int totalRotations;
+        private KCLAppConfig testConfig;
```

Comment:
+`final`?

Reply:
totalRotations gets updated, but I can change testConfig to be final